File: packages/shared-types/python/src/mycelium_shared_types/workflow.py
"""WorkflowState — current state of a multi-step workflow."""

from __future__ import annotations

from datetime import datetime, timezone
from enum import Enum
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict, Field


class WorkflowStatus(str, Enum):
    PENDING = "pending"
    RUNNING = "running"
    AWAITING_APPROVAL = "awaiting_approval"
    APPROVED = "approved"
    REJECTED = "rejected"
    COMPLETED = "completed"
    FAILED = "failed"


class WorkflowState(BaseModel):
    model_config = ConfigDict(extra="forbid")

    workflow_id: str
    name: str
    status: WorkflowStatus = WorkflowStatus.PENDING
    correlation_id: str
    parent_correlation_id: Optional[str] = None
    steps_completed: list[str] = Field(default_factory=list)
    next_step: Optional[str] = None
    context: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

File: packages/shared-types/typescript/src/workflow.ts
export type WorkflowStatus =
  | "pending"
  | "running"
  | "awaiting_approval"
  | "approved"
  | "rejected"
  | "completed"
  | "failed";

export interface WorkflowState {
  workflow_id: string;
  name: string;
  status: WorkflowStatus;
  correlation_id: string;
  parent_correlation_id?: string | null;
  steps_completed: string[];
  next_step?: string | null;
  context: Record<string, unknown>;
  /** ISO 8601 timestamps */
  created_at: string;
  updated_at: string;
}
